plugins/inputs/minecraft/internal/rcon/rcon.go

```diff
@@ -67,17 +67,17 @@ func (p Packet) Compile() (payload []byte, err error) {
 	var padding [PacketPaddingSize]byte
 
 	if err = binary.Write(&buffer, binary.LittleEndian, &size); nil != err {
-		return
+		return nil, err
 	} else if err = binary.Write(&buffer, binary.LittleEndian, &p.Header.Challenge); nil != err {
-		return
+		return nil, err
 	} else if err = binary.Write(&buffer, binary.LittleEndian, &p.Header.Type); nil != err {
-		return
+		return nil, err
 	}
 
-	if _, err := buffer.WriteString(p.Body); err != nil {
+	if _, err = buffer.WriteString(p.Body); err != nil {
 		return nil, err
 	}
-	if _, err := buffer.Write(padding[:]); err != nil {
+	if _, err = buffer.Write(padding[:]); err != nil {
 		return nil, err
 	}
```

Comment:
You probably should use local errors with `if err := ...; err != nil {` here concisely?

Reply:
I can, but since there is named return "argument" called `err`, it can be reused everywhere. Or not?

Reply:
I agree.